### PR TITLE
Added variable KAFKA_COMPRESSION_TYPE to optionally add compression type when sending Metastore events to Kafka

### PR DESCRIPTION
**Pull request by javsanbel2**
### :pencil: Description
Added variable `KAFKA_COMPRESSION_TYPE` to optionally add compression type when sending Metastore events to Kafka through apiary-metastore-listener library.

### CODE REVIEW

File: files/startup.sh [hunk 1]
```diff
@@ -104,6 +104,7 @@ if [[ ! -z $KAFKA_BOOTSTRAP_SERVERS ]]; then
     [[ -n $ECS_CONTAINER_METADATA_URI ]] && export KAFKA_CLIENT_ID=$(wget -q -O - ${ECS_CONTAINER_METADATA_URI}/task|jq -r .TaskARN|awk -F/ '{ print $NF }')
     [[ -n $KUBERNETES_SERVICE_HOST ]] && export KAFKA_CLIENT_ID="$HOSTNAME"
     [[ -n $KAFKA_CLIENT_ID ]] && sed "s/KAFKA_CLIENT_ID/$KAFKA_CLIENT_ID/" -i /etc/hive/conf/hive-site.xml
+    [[ -n $KAFKA_COMPRESSION_TYPE ]] && update_property.py com.expediagroup.apiary.extensions.events.metastore.kafka.messaging.compression.type "gzip" /etc/hive/conf/hive-site.xml
```
[the remark this answers is not in the document]
Comment (javsanbel2) [reply files/startup.sh:107]: You are right, I am not sure why I added "gzip" directly, thank you

File: README.md [hunk 2]
```diff
@@ -43,6 +43,9 @@ For more information please refer to the main [Apiary](https://github.com/Expedi
 |RANGER_SERVICE_NAME|No|Ranger service name used to configure RangerAuth plugin.|
 |SNS_ARN|No|The SNS topic ARN to which metadata updates will be sent.|
 |TABLE_PARAM_FILTER|No|A regular expression for selecting necessary table parameters. If the value isn't set, then no table parameters are selected.|
+|KAFKA_BOOTSTRAP_SERVERS|No|Kafka Bootstrap Servers to enable Kafka Metastore listener and send Metastore events to Kafka.|
```
Comment (abhimanyugupta07) [inline README.md:46]: Thanks for adding the missing entries. please move these below `INSTANCE_NAME` as they are in alphabetical order.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,10 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html). 
 
+## [3.0.11] - 2023-07-25
+### Added
+- Added variable `KAFKA_COMPRESSION_TYPE` to optionally add compression type when sending Metastore events to Kafka through apiary-metastore-listener library.
```
Comment (abhimanyugupta07) [inline CHANGELOG.md:8]: changelog will need entry for version upgrades.

**Review by abhimanyugupta07 — APPROVED** [review]

File: files/startup.sh [hunk 4]
```diff
@@ -104,6 +104,7 @@ if [[ ! -z $KAFKA_BOOTSTRAP_SERVERS ]]; then
     [[ -n $ECS_CONTAINER_METADATA_URI ]] && export KAFKA_CLIENT_ID=$(wget -q -O - ${ECS_CONTAINER_METADATA_URI}/task|jq -r .TaskARN|awk -F/ '{ print $NF }')
     [[ -n $KUBERNETES_SERVICE_HOST ]] && export KAFKA_CLIENT_ID="$HOSTNAME"
     [[ -n $KAFKA_CLIENT_ID ]] && sed "s/KAFKA_CLIENT_ID/$KAFKA_CLIENT_ID/" -i /etc/hive/conf/hive-site.xml
+    [[ -n $KAFKA_COMPRESSION_TYPE ]] && update_property.py com.expediagroup.apiary.extensions.events.metastore.kafka.messaging.compression.type "$KAFKA_COMPRESSION_TYPE" /etc/hive/conf/hive-site.xml
```
Comment (abhimanyugupta07) [inline files/startup.sh:107]: not sure if it has any impact, the other usages in this file are putting the variable as - "${HMS_MAX_THREADS}"

**Pull request merged**